tests/features/test_features.py

```diff
@@ -244,6 +244,23 @@ def test_encode_nested_example_sequence_with_none():
     assert result is None
 
 
+def test_encode_batch_with_example_with_empty_first_elem():
+    features = Features(
+        {
+            "x": Sequence(Sequence(ClassLabel(names=["a", "b"]))),
+        }
+    )
+    encoded_batch = features.encode_batch(
+        {
+            "x": [
```

Comment:
To be properly tested, does the first element has to be an empty list ?

Reply:
Yes, because `encode_batch` calls `encode_nested_example` which goes from the beginning of the list and tries to find the first element that is "good enough" to perform additional checks, and we consider an element "good enough" if it's not `None` or if it's not an empty sequence (possible nested). Previously, we would stop on the first element that is not `None`, but this could lead to issues such as the one this PR fixes.